src/react/features/tools/delete/useDeleteDioryAndLinks.js

```diff
@@ -0,0 +1,66 @@
+import { useStore, useDispatch } from '../../../store'
+import { setOpen, setInactive } from '../../buttons/actions'
+import { goBackward, setSelectedLink } from '../../navigation/actions'
+import { deleteDiory, deleteLink } from '../../diograph/actions'
+
+const generateDeletedLinks = (focus, links, isFocusDeleted) => {
+  const [{ diograph }] = useStore((state) => state.diograph)
+
+  const linkedDiories = Object.values(links).map(({ id }) => ({
+    fromDiory: diograph[focus.id],
+    toDiory: diograph[id],
+  }))
+
+  if (isFocusDeleted) {
+    const reverseLinkedDiories = Object.values(diograph)
+      .map((diory) => {
+        const reverseLinkedDiory = diory.links
+          ? Object.entries(diory.links).filter(([, { id }]) => id === focus.id)[0]
+          : null
+        return reverseLinkedDiory
+          ? {
+              fromDiory: diograph[diory.id],
+              toDiory: diograph[focus.id],
+            }
+          : null
+      })
+      .filter(Boolean)
+    return linkedDiories.concat(reverseLinkedDiories)
+  }
+
+  return linkedDiories
+}
+
+export const useDeleteDioryAndLinks = (focus, clickedDiory) => {
+  const isFocusDeleted = focus.id === clickedDiory.id
+  const deletedDiory = isFocusDeleted ? focus : null
+  const links = isFocusDeleted ? focus.links || {} : { [clickedDiory.id]: { id: clickedDiory.id } }
+  const deletedLinks = generateDeletedLinks(focus, links, isFocusDeleted)
```

Comment:
One of redux style guides is:
"Put as Much Logic as Possible in Reducers#
Wherever possible, try to put as much of the logic for calculating a new state into the appropriate reducer, rather than in the code that prepares and dispatches the action"

According to that style guide, the most this logic should be in a reducer.

Reply:
I will try this tomorrow as I also make a new suggestion from https://github.com/DioryMe/diory-browser-electron/pull/111 . Need to do this kind of stuff there too...

Reply:
Thanks for the style guide quote, didn't know that!

I refactored the whole thing and it came out so much clearer! (although also took a good hour to do...)
Didn't end up putting the logic into the reducer because our DeleteView needs the list of deleted links too. Otherwise it would be a good idea (and also pretty trivial task after the refactor...) to put the logic there. And maybe if we need to delete stuff elsewhere we have to consider make own helpers for the deleting stuff...

(in fact, don't we currently already have a delete tool on map... 🤔 )